File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.java
class Solution {
    public int[] vowelStrings(String[] words, int[][] queries) {
        HashSet<Character> hs = new HashSet<>();
        hs.add('a');
        hs.add('e');
        hs.add('i');
        hs.add('o');
        hs.add('u');
        /*boolean[] arr = new boolean[words.length];
        for(int i=0;i<words.length;i++){
            String word = words[i];
            int wordlength = word.length();
            if(hs.contains(word.charAt(0)) && hs.contains(word.charAt(wordlength-1))){
                arr[i] = true;
            }
            else{
                arr[i] = false;
            }
        }
        int[] ans = new int[queries.length];
        for(int i=0;i<queries.length;i++){
            int count = 0;
            int start = queries[i][0];
            int end = queries[i][1];
            for(int j=start;j<=end;j++){
                if(arr[j]==true){
                    count++;
                }
            }
            ans[i] = count;
        }
        return ans;*/
        int n = words.length;
        int[] prefix = new int[n+1];
        for(int i=0;i<n;i++){
            String word = words[i];
            int wordlength = word.length();
            boolean isvowelstring = hs.contains(word.charAt(0)) && hs.contains(word.charAt(wordlength-1));
            prefix[i+1] = prefix[i]+(isvowelstring?1:0);
        }
        int[] ans = new int[queries.length];
        for(int i=0;i<queries.length;i++){
            int start = queries[i][0];
            int end = queries[i][1];
            ans[i] = prefix[end+1]-prefix[start];
        }
        return ans;
    }
}

File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        vowels = {'a','e','i','o','u'}
        n = len(words)
        prefix = [0]*(n+1)
        for i in range(n):
            word = words[i]
            isvowelstring = word[0] in vowels and word[-1] in vowels
            prefix[i+1] = prefix[i]+(1 if isvowelstring else 0)
        ans=[]
        for start,end in queries:
            ans.append(prefix[end+1]-prefix[start])
        return ans
